### PR TITLE
Fix apgar 2

### PR DESCRIPTION
**Pull request by pieterbos**
same as the one before, just rebased

### CODE REVIEW

File: aom/src/main/java/com/nedap/archie/query/AOMPathQuery.java [hunk 1]
```diff
@@ -25,11 +25,20 @@
  */
 public class AOMPathQuery {
 
-    private List<PathSegment> pathSegments = new ArrayList<>();
+    private final List<PathSegment> pathSegments;
+
+    /** If true, extend the search through C_COMPLEX_OBJECT_PROXY objects by looking up the replacement first.*/
+    private final boolean findThroughCComplexObjectProxies;
```
Comment (steijgeler) [inline aom/src/main/java/com/nedap/archie/query/AOMPathQuery.java]: So when should or shouldn't the proxy be replaced when doing an apathQuery? Seems to me it should always replace it?

File: tools/src/main/java/com/nedap/archie/archetypevalidator/validations/BasicTerminologyValidation.java [hunk 2]
```diff
@@ -68,7 +69,7 @@ private void validateTerminologyBindings() {
             for(String constraintCodeOrPath: termBindings.get(terminologyId).keySet()) {
                 boolean archetypeHasPath = false;
                 try {
-                    archetypeHasPath = archetype.hasPath(constraintCodeOrPath);
+                    archetypeHasPath = !new AOMPathQuery(constraintCodeOrPath).findThroughCComplexObjectProxies().findList(archetype.getDefinition()).isEmpty();//archetype.hasPath(constraintCodeOrPath);
```
Comment (steijgeler) [inline tools/src/main/java/com/nedap/archie/archetypevalidator/validations/BasicTerminologyValidation.java]: comment can be removed

**Comment by wolandscat** [comment]
A path query should always resolve down a proxy object target - whether you achieve that by an actual replacement, or a different means of computation, is a choice. I don't do replacements until flattening in the ADL Workbench, but they could be done earlier potentially.

**Comment by pieterbos** [comment]
Thanks for the clarification. It has been fixed. It does not do replacements before flattening, but does a calculation using the ComplexObjectProxyReplacement as a helper class.

**Pull request merged**